lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -2010,7 +2010,10 @@ Attribute DotOperandEncodingAttr::parse(AsmParser &parser, Type type) {
   unsigned kWidth = 0;
   Attribute _kWidth = attrs.get("kWidth");
   if (_kWidth) {
-    if (!mmaParent || mmaParent.isVolta()) {
+    bool isAmdWithKWidth =
+        parent.isa<AMDMfmaEncodingAttr>() || parent.isa<AMDWmmaEncodingAttr>();
```

Comment:
Upstream mlir is moving to `isa<..Attr>()` style and may deprecate this one later. For future proof, let's do that right now.